Gallery finished

Samuel-Fox opened the pull request:
# Description

Added a gallery screen that lets the users rotate through the different pieces in the game along with a short description and artist credits to accompany each piece.

## Related issue(s)

#11 

## List of changes

A more detailed list of the changes.
- Added the Gallery as a new scene
- Gallery button from main screen switches to the gallery when clicked
- Next and Previous buttons loop users through an array on pieces and a description of the piece

## Additional notes

Open to changing UI if some other layout of the elements is preferable. Do a double check on grammar of descriptions and make sure I did not miscredit any artists, I made a list to make sure each artist got the proper credit but it's not impossible I missed something.

Mikotey commented on Birthday-2024-Project/Scripts/UI/MainMenu/MainMenuUiManager.gd:
```diff
@@ -41,11 +42,15 @@ func show_campaign_levels():
+	_disable_all_screans()
+	gallery_screen.show()
 
 func _disable_all_screans():
```
I never noticed this said "screans" lol

Mikotey reviewed (commented):
Screen looks good! I like the placements of the pieces and text. All the buttons are setup and working properly.

@Casulono We might need to review the piece descriptions. They seem to be missing articles (a, an, the) in parts. And it feels like some have run on sentences.

This PR might have to wait until the descriptions get updated.
Or, since all the logic is in place, we could push the PR and update the text later with another PR. This PR uses the older style of main menu screen management. Thus, I'll probably need to merge it into my PR and resolve the conflicts

![image](https://github.com/Saplings-Projects/Birthday-2024/assets/9286524/9e75faca-c616-4f39-8cdc-8dab6bc56cdf)
There are these arrow assets (under Art/UI) that I started using for the level select screens. I think it would look nice if we swapped the previous and next buttons with these arrows. What do you think?
You can checkout the implementation in MenuRefactorStates PR branch if you want to interact with the buttons

Casulono commented:
@Samuel-Fox @Mikotey Went through several online checkers and also did a manual one. Here is the result: https://docs.google.com/document/d/1qsYnMhI8IoCFPuJJNmh6ylV0C0Hmzwu9wFTi8Myj2A0/edit

Casulono commented on Birthday-2024-Project/ScenePrefabs/gallery.tscn:
```diff
@@ -0,0 +1,370 @@
+[gd_scene load_steps=31 format=3 uid="uid://k787vo3otomq"]
```
minor thing: change the name of the scene to "Gallery" to have it consistent with other scenes

Samuel-Fox commented:
Pushed new changes

- Added name of the pieces above the descriptions
- Update descriptions from the google doc
- Changed UI to more closely match the one from the level selection
- Fixed capitalization on gallery

Casulono commented on Birthday-2024-Project/ScenePrefabs/MainMenuUiManager.tscn:
```diff
@@ -1,12 +1,13 @@
 [ext_resource type="PackedScene" uid="uid://4kktp1gwthbt" path="res://ScenePrefabs/CampaignSelectMenu.tscn" id="3_qksbd"]
 [ext_resource type="Script" path="res://Scripts/UI/MainMenu/States/MainMenuUiStartState.gd" id="3_vnjds"]
 [ext_resource type="PackedScene" uid="uid://lma3uuskmlpo" path="res://ScenePrefabs/CampaignLevelSelectMenu.tscn" id="4_o3atr"]
+[ext_resource type="PackedScene" uid="uid://k787vo3otomq" path="res://ScenePrefabs/Gallery.tscn" id="5_1folx"]
```
actual gallery.tscn file is still without a capital letter

Samuel-Fox replied:
Should be fixed now

Casulono commented on Birthday-2024-Project/ScenePrefabs/gallery.tscn:
```diff
@@ -0,0 +1,400 @@
+piece = NodePath("Piece Display/Casual Fauna Wave")
+pieces = [NodePath("Piece Display/Casual Fauna Wave"), NodePath("Piece Display/Casual Faunya"), NodePath("Piece Display/Faunya Hungry"), NodePath("Piece Display/Happy Fauna"), NodePath("Piece Display/Apple Blocker"), NodePath("Piece Display/Maid Fauna"), NodePath("Piece Display/Nemu Fauna"), NodePath("Piece Display/Casual Fauna"), NodePath("Piece Display/Faunashi"), NodePath("Piece Display/Big Ma ...
+description = NodePath("Text Display/Piece Description")
+descriptions = Array[String](["\"Konfauna! I’m Ceres Fauna from Hololive Promise, here to make you return to nature as all humans should.”", "Is it a cat? Is it Casual Fauna? It’s actually both! And it seems she likes this golden apple she has in her paws. Don’t try to steal it from her.", "It’s surprising how much food this smol Faunya can eat. It’s almost like she's a bottomless pit. And now sh ...
```
"she will provide commentary for the participants in the GoI Trials! for the participants of GOI Trials!" remove one instance of "for the participants of GOI Trials!"

Casulono replied:
I've updated in doc "is it possible to overdose on it?" to "is it possible to overdose it?". Should also be pulled here

Samuel-Fox replied:
Fixed both instances of incorrect descriptions
